Rewrite concepts section (#203)
onflow/docs: docs/concepts/events.md

---
sidebar_position: 6
---

# Events

> Flow events are special values that can be emitted during the execution of a Cadence program. 

Reviewer: ```suggestion
Flow events are special values that are emitted to off-chain chain observers during the execution of a Cadence program. 
```

| --- | --- |
| flow.AccountCreated | Event that is emitted when a new account gets created. |
| flow.AccountKeyAdded  | Event that is emitted when a key gets added to an account. |
| flow.AccountKeyRemoved  | Event that is emitted when a key gets removed from an account. |
| flow.AccountContractAdded  | Event that is emitted when a contract gets deployed to an account. |
| flow.AccountContractUpdated | Event that is emitted when a contract gets updated on an account. |
| flow.AccountContractRemoved  | Event that is emitted when a contract gets removed from an account. |
| flow.InboxValuePublished | Event that is emitted when a Capability is published from an account. |
| flow.InboxValueUnpublished | Event that is emitted when a Capability is unpublished from an account. |
| flow.InboxValueClaimed1 | Event that is emitted when a Capability is claimed by an account. |

For more details [on the core events, you can read Cadence reference documentation](https://developers.flow.com/cadence/language/core-events).

## User-defined events

Events that are defined inside contracts and when emitted follow a common naming schema. The schema consists of 4 parts: 

> ```json

Reviewer 2: use correct language

A.{contract address}.{contract name}.{event type}
```

An example event would look like:

![Screenshot 2023-08-18 at 14.30.36.png](_events_images/Screenshot_2023-08-18_at_14.30.36.png)

The first `A` means the event is originating from a contract, which will always be the case for user-defined events. The contract address as the name implies is the location of a contract deployed on the Flow network. Next is the name of the contracted event originates from, and last is the event type defined in the contract.

There are an unlimited amount of events that can be defined on Flow, but you should know about the most common ones. 

### Flow Token Events

> The Flow Token contract is the contract that issues a core Flow token. As with any contract, it can emit events when interacted with. When we transfer the Flow token, events are emitted. You can find a lot of details on the events emitted in the [Flow Token documentation](../concepts/core-contracts/03-flow-token.md). 

Reviewer 3: Recommendation to use upper case FLOW if talking specifically about the native FLOW token. The nomenclature is "Flow" talking about the chain and "FLOW" talking about the native FLOW token.
